addons/info/src/components/types/ObjectType.js

```diff
@@ -0,0 +1,79 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import PrettyPropType from './PrettyPropType';
+import PropertyLabel from './PropertyLabel';
+import HighlightButton from './HighlightButton';
+
+import { TypeInfo } from './proptypes';
+
+const MARGIN_SIZE = 15;
+
+export default class ObjectType extends React.Component {
+  static propTypes = {
+    propType: TypeInfo,
+    depth: PropTypes.number.isRequired,
+  };
+
+  static defaultProps = {
+    propType: null,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      minimized: false,
+    };
+  }
+
+  handleToggle = () => {
+    this.setState({
+      minimized: !this.state.minimized,
+    });
+  };
+
+  handleMouseEnter = () => {
```

Comment:
Please convert to glamorous component to do hover styling.

Reply:
In this case, the hover state is required because I highlight the matching bracket.